### ISSUE TITLE
Chain verification missing

### ISSUE THREAD

**Issue by yashasvi-ranawat** [pybitcash/bitcash#133]
A wallet does not check that the chain (mainnet, testnet or regtest) the output addresses are on matches the one the wallet is on, before sending a transaction.

**Comment by merc1er** [comment]
Does this result in the transaction being rejected by the network?

**Comment by yashasvi-ranawat** [comment]
yes! if you intend to spend a tx on testnet, but accidentally use a mainnet wallet, it won't find the utxos; or worse, send real BCH to a wallet with locking script same as the testnet address.

I saw the Bit repo, it does have a check for such case.
